modules/assignment/src/main/java/com/intuit/wasabi/assignment/AssignmentDecorator.java

```diff
@@ -15,23 +15,38 @@
  *******************************************************************************/
 package com.intuit.wasabi.assignment;
 
-import com.intuit.wasabi.experimentobjects.Experiment;
+import java.io.IOException;
 
-import java.io.UnsupportedEncodingException;
-import java.net.URI;
+import com.intuit.wasabi.assignmentobjects.SegmentationProfile;
+import com.intuit.wasabi.assignmentobjects.User;
+import com.intuit.wasabi.exceptions.BucketDistributionNotFetchableException;
+import com.intuit.wasabi.experimentobjects.BucketList;
+import com.intuit.wasabi.experimentobjects.Experiment;
 
 /**
- * Decorator for creating uri from experiment
+ * Interface that provides method(s) to modify/enhance the default behavior of retrieving bucket list when
```

Comment:
@mans4singh probably not. Today this interface AssignmentDecorator can "decorate" bucket list. Tomorrow, we can add another method to "decorate" the assignment. Let's keep the same name for now. agreed?